src/main/frontend/pipeline-graph-view/pipeline-graph/main/support/connections.tsx

```diff
@@ -0,0 +1,358 @@
+import * as React from 'react';
+
+import { nodeStrokeWidth } from '../support/StatusIcons';
+import {
+    CompositeConnection,
+    LayoutInfo,
+    NodeInfo,
+} from '../PipelineGraphModel';
+import { sequentialStagesLabelOffset } from '../PipelineGraphLayout';
+
+type SVGChildren = Array<any>; // Fixme: Maybe refine this? Not sure what should go here, we have working code I can't make typecheck
+
+// Generate a react key for a connection
+function connectorKey(leftNode: NodeInfo, rightNode: NodeInfo) {
+    return 'c_' + leftNode.key + '_to_' + rightNode.key;
+}
+
+interface Props {
+  	connections: Array<CompositeConnection>;
+    layout: LayoutInfo;
+}
+
+export class GraphConnections extends React.Component {
+    props!: Props;
+
+     /**
+     * Generate SVG for a composite connection, which may be to/from many nodes.
+     *
+     * Farms work out to other methods on self depending on the complexity of the line required. Adds all the SVG
+     * components to the elements list.
+     */
+    private renderCompositeConnection(connection: CompositeConnection, svgElements: SVGChildren) {
+        const { sourceNodes, destinationNodes, skippedNodes, hasBranchLabels } = connection;
+
+        if (skippedNodes.length === 0) {
+            // Nothing too complicated, use the original connection drawing code
```

Comment:
this comment needed?

Reply:
I'd rather not touch this, as it belongs to the original rendering code and It's a huge blackbox to me.